lib/ohai/plugins/darwin/hardware.rb

```diff
@@ -0,0 +1,102 @@
+#
+# Author:: Nate Walck (<nate.walck@gmail.com>)
+# Copyright:: Copyright (c) 2016-present Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+Ohai.plugin(:Hardware) do
+  provides 'hardware'
+
+  def system_profiler(datatype)
+    sp_cmd = "system_profiler #{datatype} -xml"
+    # Hardware queries
+    sp_std = shell_out(sp_cmd)
+    if sp_std.exitstatus == 0
+      sp_hash = Plist.parse_xml(sp_std.stdout)
+    else
+      Ohai::Log.debug("system_profiler failed to run")
```

Comment:
As a note, the shell_out method has some debugging which may duplicate this message: https://github.com/chef/ohai/blob/master/lib/ohai/mixin/command.rb#L39. It's your decision.